types/list_test.go

```diff
@@ -153,20 +153,48 @@ func TestListTypeValueFromTerraform(t *testing.T) {
 				},
 			},
 		},
+		"wrong-type": {
```

Comment:
Unrelated to the fix here, but I added these tests for consistency with objects and maps, since they were missing from lists and sets.

Reply:
Thank you!